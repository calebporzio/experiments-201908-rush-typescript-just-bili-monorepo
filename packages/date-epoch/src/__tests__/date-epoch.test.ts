/* eslint-env jest */

import { DateEpoch } from '..'

describe('DateEpoch', () => {
  const inputEpochInt = 1533762055251
  const expectEpochIntToISOString = '2018-08-08T21:00:55.251Z'
  test('DateEpoch#constructor', () => {
    const subject = new DateEpoch(inputEpochInt)
    expect(JSON.stringify(subject)).toBe(
      JSON.stringify({ epoch: inputEpochInt }),
    )
    expect(JSON.parse(JSON.stringify(subject))).toMatchObject({
      epoch: inputEpochInt,
    })
    expect(String(subject)).toEqual(inputEpochInt.toString())
    expect(subject.toDate().toISOString()).toEqual(expectEpochIntToISOString)
    expect(JSON.stringify(subject)).toMatchSnapshot()
  })

  test('DateEpoch#constructor when empty', () => {
    const subject = new DateEpoch()
    expect(subject.epoch).toBeGreaterThan(inputEpochInt)
  })

  test('DateEpoch#toString', () => {
    const epoch = 1533762055251
    const subject = new DateEpoch(epoch)
    const stringified = String(subject)
    expect(stringified).toBe(String(epoch))
  })
})
