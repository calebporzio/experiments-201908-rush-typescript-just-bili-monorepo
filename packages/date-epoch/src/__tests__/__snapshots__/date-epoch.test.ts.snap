// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateEpoch DateEpoch#constructor 1`] = `"{\\"epoch\\":1533762055251}"`;
